engine/gameobject/src/dmsdk/gameobject/gameobject.h

```diff
@@ -243,17 +245,42 @@ namespace dmGameObject
         UPDATE_RESULT_UNKNOWN_ERROR = -1000,
     };
 
+    /*# Type of property value
+     *
+     * Type of property value
+     *
+     * @enum
+     * @name PropertyValueType
+     * @member dmGameObject::PROP_VALUE_VAR
+     * @member dmGameObject::PROP_VALUE_ARRAY
+     * @member dmGameObject::PROP_VALUE_HASHTABLE
+     */
+    enum PropertyValueType
+    {
+        PROP_VALUE_VARIABLE = 0,
+        PROP_VALUE_ARRAY = 1,
+        PROP_VALUE_HASHTABLE = 2,    
+    };
```

Comment:
We wanna know property type to provide better errors handling

Reply:
We generally use 0 for the default (most common) option. Which is "arrays" in our case (i.e. variables)

Reply:
We have error mesage that uses this value:

```
                if (index_requested && (property_desc.m_ValueType != dmGameObject::PROP_VALUE_ARRAY))
                {
                    return luaL_error(L, "Options table contains index, but property '%s' is not an array.", dmHashReverseSafe64(property_id));
                }
```

Reply:
Yes, and you need to refactor such code to fit our new code?

Reply:
E:g. what are the possible values of "property_desc.m_ValueType" ?